src/datachain/lib/arrow.py

```diff
@@ -25,7 +27,17 @@
 DATACHAIN_SIGNAL_SCHEMA_PARQUET_KEY = b"DataChain SignalSchema"
 
 
+class ReferenceFileSystem(fsspec.implementations.reference.ReferenceFileSystem):
+    def _open(self, path, mode="rb", *args, **kwargs):
+        # `fsspec`'s `ReferenceFileSystem._open` reads the whole file in-memory.
```

Comment:
@skshetry what are the practical implications of this? does it mean we can't handle large files anymore?

Reply:
The comment is explaining the rationale why I am implementing `_open` method.
The `fsspec.ReferenceFileSystem._open` reads the whole file in memory and gives you a `io.BytesIO()` object. I have _fixed_ it to return a file object. 

So this should now be able to handle large files.